src/librustc_mir/borrow_check/mod.rs

```diff
@@ -830,6 +835,37 @@ impl InitializationRequiringAction {
     }
 }
 
+/// A simple linked-list threaded up the stack of recursive calls in `visit_terminator_drop`.
```

Comment:
cc @nikomatsakis this is starting to quickly become the rustc team's favorite hack!

Reply:
Indeed, in the [commit message](https://github.com/rust-lang/rust/pull/52782/commits/31c05979f935aeeabe99712e862c7affb821561b) I even wrote: 
```
Note: A similar style stack-only linked-list definition can be found
in `rustc_mir::borrow_check::places_conflict`. It might be good at
some point in the future to unify the two types and put the resulting
definition into `librustc_data_structures/`.
```

Reply:
back in my day... https://github.com/rust-lang/rust/pull/12162/files#diff-87946a10679226d1e5b4d6f9c30ab133R47